lib/chef/util/dsc/configuration_generator.rb

```diff
@@ -59,6 +57,49 @@ def configuration_document_from_script_path(script_path, configuration_name, con
 
     protected
 
+    def validate_switch_name!(switch_parameter_name)
+      if !!(switch_parameter_name =~ /\A[A-Za-z]+[_a-zA-Z0-9]*\Z/) == false
```

Comment:
```suggestion
      unless switch_parameter_name =~ /\A[A-Za-z]+[_a-zA-Z0-9]*\Z/
```

Reply:
Since we don't care about the match group we should just use .match? here and avoid a few k of memory usage